frame/uniques/src/lib.rs

```diff
@@ -0,0 +1,1289 @@
+use codec::{Encode, Decode, HasCompact};
+use frame_support::traits::{Currency, ReservableCurrency, BalanceStatus::Reserved};
+use frame_system::Config as SystemConfig;
+
+pub use weights::WeightInfo;
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T, I = ()>(_);
+
+	#[pallet::config]
+	/// The module configuration trait.
+	pub trait Config<I: 'static = ()>: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self, I>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Identifier for the class of asset.
+		type ClassId: Member + Parameter + Default + Copy + HasCompact;
+
+		/// The type used to identify a unique asset within an asset class.
+		type InstanceId: Member + Parameter + Default + Copy + HasCompact + From<u16>;
+
+		/// The currency mechanism, used for paying for reserves.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		/// The origin which may forcibly create or destroy an asset or otherwise alter privileged
+		/// attributes.
+		type ForceOrigin: EnsureOrigin<Self::Origin>;
+
+		/// The basic amount of funds that must be reserved for an asset class.
+		type ClassDeposit: Get<DepositBalanceOf<Self, I>>;
+
+		/// The basic amount of funds that must be reserved for an asset instance.
+		type InstanceDeposit: Get<DepositBalanceOf<Self, I>>;
+
+		/// The basic amount of funds that must be reserved when adding metadata to your asset.
+		type MetadataDepositBase: Get<DepositBalanceOf<Self, I>>;
+
+		/// The basic amount of funds that must be reserved when adding an attribute to an asset.
+		type AttributeDepositBase: Get<DepositBalanceOf<Self, I>>;
+
+		/// The additional funds that must be reserved for the number of bytes store in metadata,
+		/// either "normal" metadata or attribute metadata.
+		type DepositPerByte: Get<DepositBalanceOf<Self, I>>;
+
+		/// The maximum length of data stored on-chain.
+		type StringLimit: Get<u32>;
+
+		/// The maximum length of an attribute key.
+		type KeyLimit: Get<u32>;
+
+		/// The maximum length of an attribute value.
+		type ValueLimit: Get<u32>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::storage]
+	/// Details of an asset class.
+	pub(super) type Class<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::ClassId,
+		ClassDetails<T::AccountId, DepositBalanceOf<T, I>>,
+	>;
+
+	#[pallet::storage]
+	/// The assets held by any given account; set out this way so that assets owned by a single
+	/// account can be enumerated.
+	pub(super) type Account<T: Config<I>, I: 'static = ()> = StorageNMap<
+		_,
+		(
+			NMapKey<Blake2_128Concat, T::AccountId>, // owner
+			NMapKey<Blake2_128Concat, T::ClassId>,
+			NMapKey<Blake2_128Concat, T::InstanceId>,
```

Comment:
ping @jacogr

Reply:
All ok, the API does support it (UI is lagging a bit, so don't believe it is query-able yet, there is some support, but this is will break the storage page on this entry... it is on the list TODO)